src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -577,11 +581,12 @@ def listFileBlockLocation(self, fileBlockName, dbsOnly = False, phedexNodes=Fals
                 msg += "%s\n" % str(ex)
                 raise Exception(msg)
 
-            if not blockInfo or len(blockInfo) != len(blockNames): #if we couldnt get data location from PhEDEx, try to look into origin site location from dbs
-                dbsOnly = True
-                blockNames = set(blockNames) - set(blockInfo) #get the blocks we did not find information in phedex
+            if blockInfo and len(blockInfo) == len(blockNames):
+                phedexOnly = True
```

Comment:
I also don't like to change input parameters. I do it because that's what the current version is already doing (is changing the dbsOnly input parameter), so I thought I will keep the same "philosophy". Otherwise I would introduce a new boolean variable (e.g. "allLocationsFound") to signal that all locations were successfully obtained from PhEDEx (or not) and add a check on this variable: `if not phedexonly:` --> `if not phedexonly and not allLocationsFound:`.